lib/collector.js

```diff
@@ -18,10 +18,19 @@ class DataCollector {
    */
   step(info){
```

Comment:
Perhaps just change the existing method name to stepEJS or something and have another EDR specific step function that can assume `stackTop`'s existence? 

This lets you use slightly simpler logic and could make a difference since the handler is getting called a huge number of times over the course of a long test suite.